wordpress-mobile/WordPress-iOS
# Tag and Category Taxonomy Service Additions

This PR covers a few additions made to the `PostCategoryService` and `PostTagService` over the course of working on Menus.

Also, I've added new comment revisions, nullability specifications, a couple bits of refactoring for nullability awareness, and unit tests for the REST handling of remote taxonomy.

Note: I'd like to add additional unit tests covering the XML-RPC remote taxonomy implementation but couldn't seem to work out including the XML-RPC in the unit testing suite, since it lives in a pod. @diegoreymendez any insight on that one?

Note: There are currently no users of the `PostTagService` but is being used in the upcoming Menus review.

To test:

1. Run the tests for `PostServiceRemoteRESTTests`.
2. Run the tests for `PostTagServiceTests`.
3. Run the tests for `PostCategoryServiceTests`.

Additional testing:

1. Open the editor, go to the post settings.
2. Under Categories, ensure that all of the blogs categories as listed as usual.
3. Create a new category and ensure it is also created on the site.
4. Create a new category with a parent and ensure it also created on the site under the correct parent.
5. Post a new post with selected categories and ensure they land on the site as selected.
6. Update an existing post with new category selections and ensure the changes land on the site.

Additional testing:

1. Under the Settings for a site, check that the default category for the site is selected correctly.
2. Select a new category and save the change
3. Ensure the new default category selection is reflected on the site.


Needs review: @diegoreymendez
